docs/examples/use_cases/tensorflow/efficientdet/dataset/create_coco_tfrecord.py

```diff
@@ -160,37 +153,19 @@ def create_tf_example(
             category_names.append(category_index[category_id]["name"].encode("utf8"))
             area.append(object_annotations["area"])
 
-            if include_masks:
-                run_len_encoding = mask.frPyObjects(
-                    object_annotations["segmentation"], image_height, image_width
-                )
-                binary_mask = mask.decode(run_len_encoding)
-                if not object_annotations["iscrowd"]:
-                    binary_mask = np.amax(binary_mask, axis=2)
-                pil_image = PIL.Image.fromarray(binary_mask)
-                output_io = io.BytesIO()
-                pil_image.save(output_io, format="PNG")
-                encoded_mask_png.append(output_io.getvalue())
-        feature_dict.update(
-            {
-                "image/object/bbox/xmin": tfrecord_util.float_list_feature(xmin),
-                "image/object/bbox/xmax": tfrecord_util.float_list_feature(xmax),
-                "image/object/bbox/ymin": tfrecord_util.float_list_feature(ymin),
-                "image/object/bbox/ymax": tfrecord_util.float_list_feature(ymax),
-                "image/object/class/text": tfrecord_util.bytes_list_feature(
-                    category_names
-                ),
-                "image/object/class/label": tfrecord_util.int64_list_feature(
-                    category_ids
-                ),
-                "image/object/is_crowd": tfrecord_util.int64_list_feature(is_crowd),
-                "image/object/area": tfrecord_util.float_list_feature(area),
-            }
-        )
-        if include_masks:
-            feature_dict["image/object/mask"] = tfrecord_util.bytes_list_feature(
-                encoded_mask_png
-            )
+    feature_dict.update(
+        {
+            "image/object/bbox/xmin": tfrecord_util.float_list_feature(xmin),
+            "image/object/bbox/xmax": tfrecord_util.float_list_feature(xmax),
+            "image/object/bbox/ymin": tfrecord_util.float_list_feature(ymin),
+            "image/object/bbox/ymax": tfrecord_util.float_list_feature(ymax),
+            "image/object/class/text": tfrecord_util.bytes_list_feature(category_names),
+            "image/object/class/label": tfrecord_util.int64_list_feature(category_ids),
+            "image/object/is_crowd": tfrecord_util.int64_list_feature(is_crowd),
+            "image/object/area": tfrecord_util.float_list_feature(area),
+        }
+    )
```

Comment:
Why has this been moved outside `if bbox_annotations:`?

Reply:
That's because in COCO there are pictures without any bounding boxes and without that change, DALI's tfrecord reader was failing as it was expecting to find some labels and they weren't added. With that change, we add labels with empty lists even if a picture doesn't have bounding boxes.